perf/reader.go

```diff
@@ -374,7 +378,7 @@ func (pr *Reader) ReadInto(rec *Record) error {
 			if errors.Is(err, os.ErrDeadlineExceeded) {
 				// We've hit the deadline, check whether there is any data in
 				// the rings that we've not been woken up for.
-				deadlineWasExceeded = true
+				pr.pendingErr = &FlushCompleteError{Err: os.ErrDeadlineExceeded}
```

Comment:
Like I mentioned before, the two errors are not the same. You probably want this so that checking for `flush || deadline` is just `errors.As(FlushCompleteError)`? I think `errors.Is(ErrDeadlineExceeded) || errors.Is(ErrFlushed)` is more appropriate.